cfn-resources/project/cmd/resource/resource.go

```diff
@@ -521,53 +511,73 @@ func getChangeInTeams(currentTeams []ProjectTeam, oTeams []admin.TeamRole) (newT
 	return newTeams, changedTeams, removeTeams
 }
 
-// Get difference in ApiKeys
-func getChangeInAPIKeys(groupID string, currentKeys []ProjectApiKey, oKeys []admin.ApiKeyUserDetails) (newKeys, changedKeys, removeKeys []UpdateAPIKey) {
-	for _, nKey := range currentKeys {
-		if util.IsStringPresent(nKey.Key) {
-			matched := false
-			for _, oKey := range oKeys {
-				if util.AreStringPtrEqual(nKey.Key, oKey.Id) {
-					changedKeys = append(changedKeys, UpdateAPIKey{Key: *nKey.Key, UpdatePayload: &admin.UpdateAtlasProjectApiKey{Roles: nKey.RoleNames}})
-					matched = true
-					break
-				}
-			}
-			// Add to newKeys
-			if !matched {
-				newKeys = append(newKeys, UpdateAPIKey{Key: *nKey.Key, UpdatePayload: &admin.UpdateAtlasProjectApiKey{Roles: nKey.RoleNames}})
-			}
+func readTeams(teams []ProjectTeam) []admin.TeamRole {
+	var newTeams []admin.TeamRole
+	for _, t := range teams {
+		if util.IsStringPresent(t.TeamId) {
+			newTeams = append(newTeams, admin.TeamRole{TeamId: t.TeamId, RoleNames: t.RoleNames})
 		}
 	}
+	return newTeams
+}
 
-	for _, oKey := range oKeys {
-		if util.IsStringPresent(oKey.Id) {
-			matched := false
-			for _, nKey := range currentKeys {
-				if util.AreStringPtrEqual(nKey.Key, oKey.Id) {
-					matched = true
-					break
-				}
-			}
-			if !matched {
-				for _, role := range oKey.Roles {
-					// Consider only current ProjectRoles
-					if util.AreStringPtrEqual(role.GroupId, &groupID) {
-						removeKeys = append(removeKeys, UpdateAPIKey{Key: *oKey.Id})
-					}
-				}
+func getChangeInAPIKeys(currentKeys []ProjectApiKey, previousKeys []ProjectApiKey) (newKeys, changedKeys, removeKeys []ProjectApiKey) {
+	// Create maps to efficiently check for the existence of keys by ID
+	currentKeyMap := make(map[string]ProjectApiKey)
+	oKeyMap := make(map[string]ProjectApiKey)
```

Comment:
what does this variable name mean? `o` stands for?

Reply:
haha old, ill change that to previous